### PR TITLE
Add vendor.json and jar dependency info

### PR DESCRIPTION
**Pull request by untergeek**
This adds information about `vendor.json` and jar dependencies in the gemspec file.

### CODE REVIEW

**Comment by electrical** [comment]
For jar dependencies, they require jar-dependencies gem to be part of the runtime dependencies list as well. otherwise it won't be processed.

**Comment by untergeek** [comment]
Will add that, thanks!

File: docs/asciidoc/static/include/pluginbody.asciidoc [hunk 1]
```diff
@@ -855,6 +895,24 @@ IMPORTANT: All plugins have a runtime dependency on the `logstash` core gem, and
...
+https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/logstash-output-elasticsearch.gemspec#L22-L23[Elasticsearch output plugin],
+your code may depend on a jar file.  In cases such as this, the dependency is
+added in the gemspec file in this manner:
+
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  # Jar dependencies
+  s.requirements << "jar 'org.elasticsearch:elasticsearch', '1.4.0'"
+  s.add_runtime_dependency 'jar-dependencies'
```
Comment (jordansissel) [inline docs/asciidoc/static/include/pluginbody.asciidoc]: Is this runtime dep required? JRuby ships with jar-dependencies and logstash-devtools also requires it, so plugins probably don't' need it? Right?

File: docs/asciidoc/static/include/pluginbody.asciidoc [hunk 2]
```diff
@@ -883,6 +941,17 @@ Then, you'll need to install your plugins dependencies with bundler:
 bundle install
 ----------------------------------
 
+[IMPORTANT]
+======
+
+If your plugin has an external file dependency described in `vendor.json`, you
+must download that dependency before testing:
```
Comment (jordansissel) [inline docs/asciidoc/static/include/pluginbody.asciidoc]: > must download that dependency before testing

"must download that dependency before testing or running. You can use `rake vendor` to do this.

Comment (jordansissel) [inline docs/asciidoc/static/include/pluginbody.asciidoc]: Thoughts? ^^

**Comment by jordansissel** [comment]
Other than two comments, LGTM.

File: docs/asciidoc/static/include/pluginbody.asciidoc (hunk 1 above)
Comment (untergeek) [inline docs/asciidoc/static/include/pluginbody.asciidoc]: @electrical, is this true?

Comment (jordansissel) [inline docs/asciidoc/static/include/pluginbody.asciidoc]: Ok IJ ust tested and without jar-dependencies in the runtime deps, the spec `requirements` are not activated, which is weird considering logstash requires jar-dependencies so the requirement is there, just not as a first-level thing on the plugin.

Lesson learned! `jar-dependnecies` is required on all plugins with jar deps.

**Pull request closed without merging**